src/CircleEditor/CircleType.ts

```diff
@@ -0,0 +1,316 @@
+  SVDFOPTIONS = 6,
+  RNNOPTIONS = 7,
+  FULLYCONNECTEDOPTIONS = 8,
+  SOFTMAXOPTIONS = 9,
+  CONCATENATIONOPTIONS = 10,
+  ADDOPTIONS = 11,
+  L2NORMOPTIONS = 12,
+  LOCALRESPONSENORMALIZATIONOPTIONS = 13,
+  LSTMOPTIONS = 14,
+  RESIZEBILINEAROPTIONS = 15,
+  CALLOPTIONS = 16,
+  RESHAPEOPTIONS = 17,
+  SKIPGRAMOPTIONS = 18,
+  SPACETODEPTHOPTIONS = 19,
+  EMBEDDINGLOOKUPSPARSEOPTIONS = 20,
+  MULOPTIONS = 21,
+  PADOPTIONS = 22,
+  GATHEROPTIONS = 23,
+  BATCHTOSPACENDOPTIONS = 24,
+  SPACETOBATCHNDOPTIONS = 25,
+  TRANSPOSEOPTIONS = 26,
+  REDUCEROPTIONS = 27,
+  SUBOPTIONS = 28,
+  DIVOPTIONS = 29,
+  SQUEEZEOPTIONS = 30,
+  SEQUENCERNNOPTIONS = 31,
+  STRIDEDSLICEOPTIONS = 32,
+  EXPOPTIONS = 33,
+  TOPKV2OPTIONS = 34,
+  SPLITOPTIONS = 35,
+  LOGSOFTMAXOPTIONS = 36,
+  CASTOPTIONS = 37,
+  DEQUANTIZEOPTIONS = 38,
+  MAXIMUMMINIMUMOPTIONS = 39,
+  ARGMAXOPTIONS = 40,
+  LESSOPTIONS = 41,
+  NEGOPTIONS = 42,
+  PADV2OPTIONS = 43,
+  GREATEROPTIONS = 44,
+  GREATEREQUALOPTIONS = 45,
+  LESSEQUALOPTIONS = 46,
+  SELECTOPTIONS = 47,
+  SLICEOPTIONS = 48,
+  TRANSPOSECONVOPTIONS = 49,
+  SPARSETODENSEOPTIONS = 50,
+  TILEOPTIONS = 51,
+  EXPANDDIMSOPTIONS = 52,
+  EQUALOPTIONS = 53,
+  NOTEQUALOPTIONS = 54,
+  SHAPEOPTIONS = 55,
+  POWOPTIONS = 56,
+  ARGMINOPTIONS = 57,
+  FAKEQUANTOPTIONS = 58,
+  PACKOPTIONS = 59,
+  LOGICALOROPTIONS = 60,
+  ONEHOTOPTIONS = 61,
+  LOGICALANDOPTIONS = 62,
+  LOGICALNOTOPTIONS = 63,
+  UNPACKOPTIONS = 64,
+  FLOORDIVOPTIONS = 65,
+  SQUAREOPTIONS = 66,
+  ZEROSLIKEOPTIONS = 67,
+  FILLOPTIONS = 68,
+  BIDIRECTIONALSEQUENCELSTMOPTIONS = 69,
+  BIDIRECTIONALSEQUENCERNNOPTIONS = 70,
+  UNIDIRECTIONALSEQUENCELSTMOPTIONS = 71,
+  FLOORMODOPTIONS = 72,
+  RANGEOPTIONS = 73,
+  RESIZENEARESTNEIGHBOROPTIONS = 74,
+  LEAKYRELUOPTIONS = 75,
+  SQUAREDDIFFERENCEOPTIONS = 76,
+  MIRRORPADOPTIONS = 77,
+  ABSOPTIONS = 78,
+  SPLITVOPTIONS = 79,
+  UNIQUEOPTIONS = 80,
+  REVERSEV2OPTIONS = 81,
+  ADDNOPTIONS = 82,
+  GATHERNDOPTIONS = 83,
+  COSOPTIONS = 84,
+  WHEREOPTIONS = 85,
+  RANKOPTIONS = 86,
+  REVERSESEQUENCEOPTIONS = 87,
+  MATRIXDIAGOPTIONS = 88,
+  QUANTIZEOPTIONS = 89,
+  MATRIXSETDIAGOPTIONS = 90,
+  HARDSWISHOPTIONS = 91,
+  IFOPTIONS = 92,
+  WHILEOPTIONS = 93,
+  DEPTHTOSPACEOPTIONS = 94,
+  NONMAXSUPPRESSIONV4OPTIONS = 95,
+  NONMAXSUPPRESSIONV5OPTIONS = 96,
+  SCATTERNDOPTIONS = 97,
+  SELECTV2OPTIONS = 98,
+  DENSIFYOPTIONS = 99,
+  SEGMENTSUMOPTIONS = 100,
+  BATCHMATMULOPTIONS = 101,
+  CUMSUMOPTIONS = 102,
+  CALLONCEOPTIONS = 103,
+  BROADCASTTOOPTIONS = 104,
+  RFFT2DOPTIONS = 105,
+  CONV3DOPTIONS = 106,
+  HASHTABLEOPTIONS = 107,
+  HASHTABLEFINDOPTIONS = 108,
+  HASHTABLEIMPORTOPTIONS = 109,
+  HASHTABLESIZEOPTIONS = 110,
+  VARHANDLEOPTIONS = 111,
+  READVARIABLEOPTIONS = 112,
+  ASSIGNVARIABLEOPTIONS = 113,
+  RANDOMOPTIONS = 114,
+  BCQGATHEROPTIONS = 252,
+  BCQFULLYCONNECTEDOPTIONS = 253,
+  INSTANCENORMOPTIONS = 254,
+}
+
+/**
+ * map CircleType string to CircleType class.
```

Comment:
It seems that keys are not string but primitive types.
Am I right?

Reply:
I receive types in string format or enum format from webview. And, I create object dynamically using that type and maps. So, I commented the keys are string or enum. How about just type name except for string or enum?

Reply:
Hmm... my point is that `TensorType`, `DimensionType`, ... are not string but primitive type.
For example, `int` and `'int32[]'` are different type because each are primitive and string type.

Reply:
We acknowledge the comment does not match with the actual code in respect of that it says the key is string even it's not. We'll fix this and commit again!